src/odemis/gui/cont/correlation.py

```diff
@@ -238,14 +238,22 @@ def on_char(self, evt: wx.Event) -> None:
         """handle key presses
         :param evt: (wx.Event) the event"""
 
-        if not self.correlation_enabled():
-            logging.debug("correlation not enabled, passing event to canvas")
-            active_canvas = evt.GetEventObject()
-            active_canvas.on_char(evt)            # super event passthrough
-
+        # event data
         key = evt.GetKeyCode()
         shift_mod = evt.ShiftDown()
 
+        # pass event to canvas, if not a valid correlation key
+        valid_keys = [wx.WXK_LEFT, wx.WXK_RIGHT, wx.WXK_UP, wx.WXK_DOWN]
+        if evt.GetKeyCode() not in valid_keys:
+            logging.debug(f"invalid correlation key, passing event to canvas, key was: {key}")
+            active_canvas = evt.GetEventObject()
+            active_canvas.on_char(evt)
```

Comment:
Looking at this code again, I don't think this is the proper way. It works because all the canvases have a `on_char` method. But I think the right way in wxWidgets is to use evt.Skip(). It tells wxWidget that this event handler is not interested, and the next event handler should take care of it (which will be the canvas, if it has bound to that event).